ttnn/cpp/ttnn/operations/ccl/reduce_scatter/device/host/reduce_scatter_full_worker_grid.cpp

```diff
@@ -386,12 +388,12 @@ static void add_worker_config_to_edm_builders(
                 sender_edm_builder.add_sender_channel(
                     worker_sender_semaphore_id,
                     1,  // cw_edm_channel_num_messages_to_send_per_transfer.at(c) * (ring_size - 1),
-                    sender_worker_coords,
-                    expected_message_size_bytes);
+                    sender_worker_coords);
             edm_interface_addresses.worker_sender_edm_semaphore_addresses.insert(
                 {global_worker_idx, sender_channel_buffer_info.eth_semaphore_l1_address});
             edm_interface_addresses.worker_sender_edm_buffer_addresses.insert(
                 {global_worker_idx, sender_channel_buffer_info.eth_buffer_l1_address});
+            sender_edm_builder.set_max_message_size_bytes(sender_channel_buffer_info.channel, clockwise_edm_builders.at(0).get_eth_buffer_size_bytes());
```

Comment:
should be `.at(link)`?

Reply:
I think all of them have the same eth_buffer_size?

Reply:
for single link, yes, but I think this will break some cases for galaxy. Generally though, this will need to be changed to be a lookup per channel. (Can you please call that instead - or add that API if it's missing?)

Reply:
sounds good. I will apply this change